src/monospace/index.ts

```diff
@@ -0,0 +1,155 @@
+
+  // If the user authorizes a project, `userResponse` will be available
+  if ("userResponse" in getInitFirebaseRes) {
+    if (getInitFirebaseRes.userResponse.success) {
+      return getInitFirebaseRes.userResponse.projectId;
+    } else {
+      return null;
+    }
+  }
+
+  const { error } = getInitFirebaseRes;
+
+  // Wait response: User hasn’t finished the interaction yet
+  if (error === "WAITING_FOR_RESPONSE") {
+    // wait and call back
+    await new Promise((res) => setTimeout(res, POLL_USER_RESPONSE_MILLIS));
+
+    // TODO: decide how long to ultimately wait before declaring
+    // that the user is never going to respond.
+
+    return pollAuthorizedProject(rid);
+  }
+
+  // FIXME: This is not being reached as the process exits before a new call is made
+
+  // Error response: User canceled without authorizing any project
+  if (error === "USER_CANCELED") {
+    // The user hasn’t authorized any project.
+    // Display appropriate error message.
+    throw new FirebaseError("User canceled without authorizing any project");
+  }
+
+  throw new FirebaseError(`Unhandled /get-init-firebase-response error: ${error}`);
+}
+
+/**
+ * Make call to init Firebase, get request id (rid) or error
+ */
+async function initFirebase(project?: string): Promise<InitFirebaseResponse> {
+  const port = getMonospaceDaemonPort();
+  if (!port) throw new FirebaseError("Undefined MONOSPACE_DAEMON_PORT");
+
+  const initFirebaseURL = new URL(`http://localhost:${port}/init-firebase`);
+
+  if (project) {
+    initFirebaseURL.searchParams.set("known_project", project);
+  }
+
+  const initFirebaseRes = await fetch(initFirebaseURL.toString(), {
+    method: "POST",
+    headers: {
+      "Content-Type": "application/json",
+    },
+  });
+
+  const initFirebaseResponse = (await initFirebaseRes.json()) as InitFirebaseResponse;
+
+  return initFirebaseResponse;
+}
+
+/**
+ * Get response from the user - authorized project or error
+ */
+async function getInitFirebaseResponse(rid: string): Promise<GetInitFirebaseResponse> {
+  const port = getMonospaceDaemonPort();
+  if (!port) throw new FirebaseError("Undefined MONOSPACE_DAEMON_PORT");
+
+  const getInitFirebaseRes = await fetch(
+    `http://localhost:${port}/get-init-firebase-response?rid=${rid}`
+  );
+
+  const getInitFirebaseJson = (await getInitFirebaseRes.json()) as GetInitFirebaseResponse;
+
+  logger.debug(`/get-init-firebase-response?rid=${rid} response:`);
+  logger.debug(getInitFirebaseJson);
+
+  return getInitFirebaseJson;
+}
+
+/**
+ * Create a .firebaserc in the project's root with the authorized project
+ * as the default project
+ */
+function createFirebaseRc(projectDir: string, authorizedProject: string): boolean {
+  const firebaseRc = loadRC({ cwd: projectDir });
+
+  firebaseRc.addProjectAlias("default", authorizedProject);
+
+  return firebaseRc.save();
+}
+
+/**
+ * Whether this is a Monospace environment
+ */
+export async function isMonospaceEnv(): Promise<boolean> {
```

Comment:
Why promisify this? Seems very unnecessary

Reply:
tbh I'm not sure too, I followed @jamesdaniels suggestions, maybe he had some idea about checking this in some async way but it ended up being just checking for the environment variable

Reply:
the thought originally there is that @hsubox76 could add in some override logic for the VSCE that was async. AFIAK she ended up taking a different approach

Reply:
I removed the Promise from it as it doesn't need to be async anymore